Assets/Scripts/Tank and MVC/TankController.cs

```diff
@@ -55,34 +56,25 @@ public void TankMovement()
             float rotation = Input.GetAxis("Horizontal1") * tankModel.tankScriptableObject.rotationSpeed * Time.deltaTime;
             tankPrefab.transform.Translate(0, 0, forward);
             tankPrefab.transform.Rotate(0, rotation, 0);
-        }
-        if(tankPrefab.tankType == Tanktype.Enemy)
-        {
-            EnemyTankMovement();
-        }
-        
+        }    
+
     }
 
     public async void CreateBullet()
     {
         if (isBulletFired)
         {
             isBulletFired = false;
-            new BulletContoller(tankModel.tankScriptableObject.bulletScriptableObject, tankPrefab.bulletSpawnAt);
+            new BulletContoller(tankModel.tankScriptableObject.bulletScriptableObject, tankPrefab.bulletSpawnAt, tankPrefab.tankType);
```

Comment:
I think you should use the Tank Model to store the data from the Scriptable object to the model so it gives you more freedom to play and change the data for every instance according to your need.